### ISSUE TITLE
Cranelift: iconst.i128 stores the wrong value for i128 values

### ISSUE THREAD

**Issue by afonso360** [bytecodealliance/wasmtime#2906]
When implmenting i128 support for aarch64 I found that `iconst` is not correctly storing i128 values on x86_64.

It looks like the issue is that we store constants as u64's in [`machineinst/lower.rs`](https://github.com/bytecodealliance/wasmtime/blob/e676589b0c6e8228c421e18249d4635eb6c4bbe4/cranelift/codegen/src/machinst/lower.rs#L1192), which then gets wrongly [lowered](https://github.com/bytecodealliance/wasmtime/blob/e676589b0c6e8228c421e18249d4635eb6c4bbe4/cranelift/codegen/src/isa/x64/lower.rs#L1500) to i128's.

### `.clif` Test Case

```
test run
target x86_64

function %i128_const_neg_1() -> i64, i64 {
block0:
    v1 = iconst.i128 -1
    v2, v3 = isplit v1
    return v2, v3
}
; run: %i128_const_neg_1() == [0xffffffff_ffffffff, 0xffffffff_ffffffff]
```

### Steps to Reproduce

`clif-util test ./the_above.clif`

### Expected Results

Returns `[-1, -1]`

### Actual Results

Returns `[-1, 0]`

### Versions and Environment

Cranelift version or commit: e676589b0c6e8228c421e18249d4635eb6c4bbe4 (main as of writing this)

Operating system: Windows 10 19042.928

Architecture: x86_64

**Comment by bjorn3** [comment]
I am suprised that the new backend supports `iconst.i128` at all. I use `iconst.i64`+`iconcat` for constructing i128 values in cg_clif.

**Comment by abrown** [comment]
I was trying to refactor the handling of immediates over in #2468, mainly from the vector perspective, but I think this could be fixed there as well. Unfortunately, I let that PR languish for a few months so I will have to revive it to fix this, unless someone here wants to pick it up.

**Comment by afonso360** [comment]
Let me try to pick that up. Lets see how it goes

**Comment by afonso360** [comment]
Fixed in #5075
